### ISSUE TITLE
App Crashes during the initial sync (Production)

### ISSUE THREAD

**Issue by AnnieMungai** [opensrp/opensrp-client-path-zeir#175]
v.2.0.2 (production)
**Device used to test: Samsung Tab A7 - Android: 9**
User: Holly
Note! This device had sufficient memory - its a new device so it doesn't have a lot of stuff on it. 
As the device was syncing data, (up to around 40pages), tried performing other tasks. 
Child registration- was successful
Advanced search - opened the advanced search page
Swiped right to access the side menu - crashed the app (gave message - ZEIR app not responding, trying to select any of the options crashed the app). 

Trying to log in with the same user crashed the app immediately it logs in. 
Clearing cache and trying to log in again still crashed the app. 
The user could not log in again successfully. 

**Testing with a different device - Xiaomi 10T pro -Android 10.** 
User: Cecilia
When I tried performing the same tasks, the app crashed the first time (not responding message). However, when I logged back in the app did not crash and sync was completed. 
![image](https://user-images.githubusercontent.com/58034047/126683467-8e3bbce7-2ebe-4af7-97e6-ffe0ad75c7e2.png)

**Testing using a different device: Huawei Mediapad T3  - Android 7**
User: Holly
App stopped responding as soon as the sync was completed. It synced to over 50pages. Logging out and trying to log in crashed the app. 
![image](https://user-images.githubusercontent.com/58034047/126780286-e0c36021-3cba-404e-82f5-3cef622a33cb.png)

**Comment by ellykits** [comment]
Possible causes for the issues reported:
- A lot of processes running on the device hence the Android OS forcing the background services running sync to stop
- App slows down with a lot of data synced from the server
- App running many tasks on the Main thread hence the reason for Device not Res

Possible fixes
- Move long running tasks to the background to avoid running so many tasks on the Main thread.
- Optimize queries used on the registers (confirm if the queries are using FTS tables)
- Resolve any reported bugs via Crashlytics - there are some common issues causing app crashes reported
- Apply any optimizations made on GIZ Malawi project @bennsimon or @ndegwamartin can provide more information

**Comment by AnnieMungai** [comment]
Additional information on this issue: 
ZEIR is currently having an issue where some devices are not able to support the application (users cannot log into the production instance and when they do, the app force closes). These devices are **Huawei media pad m2 - Android 5.1 (Lollipop) not upgradable** and **Samsung Galaxy Tab S2 8.0 – Android 5.0.2 upgradable to 6.0.1.** Is there a workaround for this for the users to still use android 5 & 6 without experiencing any challenges?

A solution could be related to [this issue](https://github.com/opensrp/opensrp-client-chw/issues/1777).

**Comment by AnnieMungai** [comment]
Comments from the meeting:
- Was not able to launch any forms on Android 5. Specific library dependency. Have to change the version. 
- Suggestion: identify why it's crashing or hanging, of it is on forms - BA can apply. 
- Sync: Let sync complete; issue still happening even when the devices are fully synced. 
- Way forward: 
 Need to figure out what's crashing on Android 5. 
 Dev to access the crashlytics for the sync issue.

**Comment by ndegwamartin** [comment]
@junaidwarsivd could you try see if [upgrading to the latest version of Evernote Jobs](https://github.com/opensrp/opensrp-client-path-zeir/blob/d06167fdca2389e97bb53a6f8c1a277c03e60d3c/opensrp-path-zeir/build.gradle#L360) fixes the crash? 

https://github.com/evernote/android-job

A possible fix might be related to this [client core issue #630](https://github.com/opensrp/opensrp-client-core/issues/630) cc @qiarie

**Comment by ndegwamartin** [comment]
@junaidwarsivd could you try see if upgrading to the latest version of Evernote Jobs fixes the crash?

https://github.com/evernote/android-job

A possible fix might be closing this client core issue #630 cc @qiarie

**Comment by ndegwamartin** [comment]
@junaidwarsivd could you try see if upgrading to the latest version of Evernote Jobs fixes the crash?

https://github.com/evernote/android-job

A possible fix might be closing this [client core issue #630](https://github.com/opensrp/opensrp-client-core/issues/630) cc @qiarie

**Comment by ndegwamartin** [comment]
@junaidwarsivd could you try see if [upgrading to the latest version of Evernote Jobs](https://github.com/opensrp/opensrp-client-path-zeir/blob/d06167fdca2389e97bb53a6f8c1a277c03e60d3c/opensrp-path-zeir/build.gradle#L360) fixes the crash?

https://github.com/evernote/android-job

A possible fix might be closing this client core issue #630 cc @qiarie

**Comment by ndegwamartin** [comment]
@junaidwarsivd could you try see if [upgrading to the latest version of Evernote Jobs](https://github.com/opensrp/opensrp-client-path-zeir/blob/d06167fdca2389e97bb53a6f8c1a277c03e60d3c/opensrp-path-zeir/build.gradle#L360) fixes the crash?

https://github.com/evernote/android-job

A possible fix might be closing this [client core issue #630](https://github.com/opensrp/opensrp-client-core/issues/630) cc @qiarie

**Comment by junaidwarsivd** [comment]
sure i am doing that @ndegwamartin

**Comment by qiarie** [comment]
@ndegwamartin The topic of the move to WorkManager came up when reviewing the issue with @ellykits. I will seek to clear the pending work on the issue.

**Comment by AnnieMungai** [comment]
@qiarie are we able to fast track this as it is affecting users on production?

**Comment by qiarie** [comment]
@AnnieMungai This will take some time to implement as it requires a rebuild of the jobs implementation with sufficient QA. It may not be enough to unblock at the moment.

**Comment by AnnieMungai** [comment]
@ellykits @junaidwarsivd After QA on the v.2.0.3 prod (tested on Android 9 device)
- [ ] The app is not exactly crashing and logging user out. However, it is displaying this message severallly. 
![image](https://user-images.githubusercontent.com/58034047/127999396-bc1618de-6c25-47b6-b253-108ced2204cb.png)
- [ ] After the initial sync is completed, when user tries to navigate through the app, they are getting different reactions. Example: Screen A was when trying to navigate from the reports page to the main register. Screen B was when I tried to get from the child registration to the main register (I did not save the form). In both scenarios, the app got to the main register after a few seconds and did not crash. 
![image](https://user-images.githubusercontent.com/58034047/127999824-50ed49f4-8ce4-42f3-981c-1c63b7857702.png)
- [ ] Opening a child profile and providing services is not creating any issues. The hanging issues are seen when the user is trying to navigate through the different parts of the app.
